avail-intellij-plugin/src/main/kotlin/org/availlang/intellij/plugin/module/AvailProjectTemplate.kt

```diff
@@ -0,0 +1,244 @@
+    val stdLib = AvailProjectRoot(
+      libConfigDir,
+      config.projectLocation,
+      libName,
+      AvailLibraries(
+        "org/availlang/${lib.name}",
+        Scheme.JAR,
+        AVAIL_STDLIB_ROOT_NAME
+      ),
+      LocalSettings(
+        AvailEnvironment.projectRootConfigPath(
+          config.fileName,
+          libName,
+          config.projectLocation
+        )
+      ),
+      sg,
+      tg,
+      extensions
+    )
+    stdLib.description = description
+    stdLib.saveLocalSettingsToDisk()
+    stdLib.saveTemplatesToDisk()
+    stdLib.saveStylesToDisk()
+    addRoot(stdLib)
+  }
+
+  /**
+   * Imports the templates from the given AvailArtifactJar and returns them as a TemplateGroup.
+   *
+   * @param config The configuration object that specifies whether to import templates.
+   * @param jar The AvailArtifactJar from which to import the templates.
+   * @return The imported templates as a TemplateGroup.
+   */
+  private fun importTemplates(
+    config: Config,
+    jar: AvailArtifactJar
+  ) = if (config.importTemplates)
+  {
+    jar.manifest.templatesFor(AVAIL_STDLIB_ROOT_NAME)
+      ?: TemplateGroup()
+  } else
+  {
+    TemplateGroup()
+  }
+
+  /**
+   * Imports the styles for the given Avail artifact jar.
+   *
+   * @param config The configuration object containing import options.
+   * @param jar The Avail artifact jar from which to import styles.
+   * @return The imported styling group if importStyles is enabled in the config object, otherwise an empty styling group.
+   */
+  private fun importStyles(
+    config: Config,
+    jar: AvailArtifactJar
+  ) = if (config.importStyles)
+  {
+    jar.manifest.stylesFor(AVAIL_STDLIB_ROOT_NAME)
+      ?: StylingGroup()
+  } else
+  {
+    StylingGroup()
+  }
+
+  /**
+   * Creates a new AvailProjectRoot object initialized with the provided parameters.
+   *
+   * @param rootConfigDir The root configuration directory.
+   * @param config The project configuration.
+   * @param rootName The root name.
+   * @param rootLocationDir The root location directory.
+   * @return The newly created AvailProjectRoot object.
+   */
+  private fun availProjectRoot(
+    rootConfigDir: String,
+    config: Config,
+    rootName: String,
+    rootLocationDir: String
+  ) = AvailProjectRoot(
+    rootConfigDir,
+    config.projectLocation,
+    rootName,
+    ProjectHome(
+      rootLocationDir,
+      Scheme.FILE,
+      config.projectLocation,
+      null
+    ),
+    LocalSettings(rootConfigDir),
+    StylingGroup(),
+    TemplateGroup()
+  )
+
+  /**
+   * Concatenates the given root directory and root name to form a new location directory.
+   *
+   * @param rootsDir The root directory string.
+   * @param rootName The root name string.
+   * @return The location directory string formed by concatenating rootsDir and rootName.
+   */
+  private fun rootsLocationDir(rootsDir: String, rootName: String) =
+    if (rootsDir.isNotEmpty()) "$rootsDir/$rootName"
+    else rootName
```

Comment:
Specify the return type. I know Kotlin can infer it, but a developer scanning the code will need to stop and do a bit more digging to know what the return type is (e.g. _hover over the function name in the IDE_). Maybe reformat it to something like:

```kotlin
private fun rootsLocationDir(
	rootsDir: String,
	rootName: String
): String =
	if (rootsDir.isNotEmpty()) "$rootsDir/$rootName"
	else rootName
```